### PR TITLE
[border-agent] update State Bitmap for ePSKc capability

### PR DESCRIPTION
**Pull request by mia1yang**
A BA at all times set the ePSKc Support flag (bit 11) in the State Bitmap (sb) TXT record of its MeshCoP DNS-SD service to '1' if the OTBR_EPSKC_ENABLE = ON;

Test: mDNS package capture - sb value changes to "00 00 08 21" from original value "00 00 00 21"

### CODE REVIEW

**Review by wgtdkp — APPROVED** [review]

File: etc/cmake/options.cmake [hunk 1]
```diff
@@ -158,3 +162,4 @@ if (OTBR_LINK_METRICS_TELEMETRY)
 else()
     target_compile_definitions(otbr-config INTERFACE OTBR_ENABLE_LINK_METRICS_TELEMETRY=0)
 endif()
+
```
Comment (bukepo) [inline etc/cmake/options.cmake:165]: unnecessary change?

File: src/border_agent/border_agent.cpp [hunk 2]
```diff
@@ -111,13 +111,19 @@ struct StateBitmap
...
 
     StateBitmap(void)
         : mConnectionMode(0)
         , mThreadIfStatus(0)
         , mAvailability(0)
         , mBbrIsActive(0)
         , mBbrIsPrimary(0)
+#if OTBR_EPSKC_ENABLE
+        , mEPSKcSupported(1)
+#endif
```
Comment (bukepo) [inline src/border_agent/border_agent.cpp:126]:
```suggestion
        , mEPSKcSupported(OTBR_EPSKC_ENABLE)
```

File: src/border_agent/border_agent.cpp [hunk 3]
```diff
@@ -111,13 +111,19 @@ struct StateBitmap
     uint32_t mAvailability : 2;
     uint32_t mBbrIsActive : 1;
     uint32_t mBbrIsPrimary : 1;
+#if OTBR_EPSKC_ENABLE
+    uint32_t mEPSKcSupported : 1;
+#endif
```
Comment (bukepo) [inline src/border_agent/border_agent.cpp:116]:
```suggestion
    uint32_t mEPSKcSupported : 1;
```

File: etc/cmake/options.cmake [hunk 4]
```diff
@@ -108,6 +108,10 @@ if(OTBR_TREL)
     target_compile_definitions(otbr-config INTERFACE OTBR_ENABLE_TREL=1)
 endif()
 
+option(OTBR_EPSKC_ENABLE "Enable ePSKc Support" OFF)
```
Comment (superwhd) [inline etc/cmake/options.cmake:111]: Per the naming convention, this should be `OTBR_EPSKC`

File: etc/cmake/options.cmake [hunk 5]
```diff
@@ -108,6 +108,10 @@ if(OTBR_TREL)
...
+if (OTBR_EPSKC_ENABLE)
+    target_compile_definitions(otbr-config INTERFACE OTBR_EPSKC_ENABLE=1)
```
Comment (superwhd) [inline etc/cmake/options.cmake:113]:
```suggestion
    target_compile_definitions(otbr-config INTERFACE OTBR_ENABLE_EPSKC=1)
```

File: src/border_agent/border_agent.cpp [hunk 6]
```diff
@@ -130,7 +136,9 @@ struct StateBitmap
         bitmap |= mAvailability << 5;
         bitmap |= mBbrIsActive << 7;
         bitmap |= mBbrIsPrimary << 8;
-
+#if OTBR_EPSKC_ENABLE
+        bitmap |= mEPSKcSupported << 11;
+#endif
```
Comment (superwhd) [inline src/border_agent/border_agent.cpp:141]:
```suggestion
        bitmap |= mEPSKcSupported << 11;
```
Similar to above comments.

File: etc/cmake/options.cmake (hunk 1 above)
Comment (mia1yang) [reply etc/cmake/options.cmake]: Thanks! Accepted.

File: etc/cmake/options.cmake (hunk 4 above)
Comment (mia1yang) [reply etc/cmake/options.cmake:111]: Thanks! Accept.

File: etc/cmake/options.cmake (hunk 5 above)
Comment (mia1yang) [reply etc/cmake/options.cmake:113]: Thanks! Accept.

File: src/border_agent/border_agent.cpp (hunk 2 above)
Comment (mia1yang) [reply src/border_agent/border_agent.cpp:126]: Accept. also need to add definition for OTBR_ENABLE_EPSK in hpp file

File: etc/cmake/options.cmake (hunk 4 above)
Comment (mia1yang) [reply etc/cmake/options.cmake:111]: Thanks! Accept.

File: src/border_agent/border_agent.cpp (hunk 6 above)
Comment (mia1yang) [reply src/border_agent/border_agent.cpp]: thanks! Accept.

File: third_party/openthread/CMakeLists.txt [hunk 7]
```diff
@@ -76,6 +76,10 @@ if (OTBR_SRP_ADVERTISING_PROXY)
     set(OT_EXTERNAL_HEAP ON CACHE STRING "enable external heap" FORCE)
 endif()
 
+if (OTBR_ENABLE_EPSKC)
+    set(OPENTHREAD_CONFIG_BORDER_AGENT_EPHEMERAL_KEY_ENABLE ON CACHE STRING "enable ePSKc" FORCE)
+endif()
```
Comment (bukepo) [inline third_party/openthread/CMakeLists.txt:81]:
```suggestion
set(OPENTHREAD_CONFIG_BORDER_AGENT_EPHEMERAL_KEY_ENABLE ON CACHE STRING "enable ePSKc" FORCE)
```

**Review by bukepo — APPROVED** [review]

File: third_party/openthread/CMakeLists.txt (hunk 7 above)
Comment (mia1yang) [reply third_party/openthread/CMakeLists.txt]: Thanks! Accept.

File: src/border_agent/border_agent.hpp [hunk 8]
```diff
@@ -65,6 +65,10 @@
 #define OTBR_MESHCOP_SERVICE_INSTANCE_NAME (OTBR_VENDOR_NAME " " OTBR_PRODUCT_NAME)
 #endif
 
+#ifndef OTBR_ENABLE_EPSKC
+#define OTBR_ENABLE_EPSKC 0
+#endif
```
[the remark this answers is not in the document]
Comment (mia1yang) [reply src/border_agent/border_agent.hpp]: Thanks! Accept.

File: src/border_agent/border_agent.cpp [hunk 9]
```diff
@@ -111,13 +111,15 @@ struct StateBitmap
     uint32_t mAvailability : 2;
     uint32_t mBbrIsActive : 1;
     uint32_t mBbrIsPrimary : 1;
+    uint32_t mEPSKcSupported : 1;
```
Comment (abtink) [inline src/border_agent/border_agent.cpp:114]: Suggest name change from `mEPSKcSupported -> mEpskcSupported`.

Our naming style uses `CamelCase` and generally applies this to abbreviations as well (treat them as a noun). e.g., `mBbrIsPrimary ` for BBR (back-bone router).

File: src/border_agent/border_agent.cpp [hunk 10]
```diff
@@ -130,7 +132,7 @@ struct StateBitmap
         bitmap |= mAvailability << 5;
         bitmap |= mBbrIsActive << 7;
         bitmap |= mBbrIsPrimary << 8;
-
+        bitmap |= mEPSKcSupported << 11;
```
Comment (abtink) [inline src/border_agent/border_agent.cpp:135]: This is okay but I am curios about using bit `11` here?  I think last one used is bit `8` for `mBbrIsPrimary`?

**Review by abtink — APPROVED** [review]
LGTM. Thanks.
Couple of smaller suggestions and question below.

File: src/border_agent/border_agent.cpp (hunk 10 above)
Comment (abtink) [reply src/border_agent/border_agent.cpp:135]: Please ignore this. Found answer already. :)

**Review by librasungirl — APPROVED** [review]

File: src/border_agent/border_agent.cpp (hunk 9 above)
Comment (mia1yang) [reply src/border_agent/border_agent.cpp]: Thanks! Accept.

**Review by superwhd — APPROVED** [review]
LGTM 👍

**Review by jwhui — APPROVED** [review]
